src/js/grid.js

```diff
@@ -62,20 +62,22 @@
     /**
      * Initializes the grid.
      */
-    init() {
-      this._node = document.getElementById("newtab-grid");
-      this._createSiteFragment();
-      addEventListener("resize", this);
+    init: async(function*() {
+      gGrid._node = document.getElementById("newtab-grid");
+      gGrid._createSiteFragment();
 
-      gNewTab.sendToBrowser("NewTab:InitializeGrid");
-      gNewTab.registerListener("NewTab:InitializeLinks",
-        this.refresh.bind(this));
+      gGrid._sw = (yield navigator.serviceWorker.ready).active;
```

Comment:
I need to come up with a better solution for how to use swMessage... it has the nice property that it's a reusable function... just need to integrate it properly into the parent's object structure... will think about that.